### ISSUE TITLE
Should we continue using the term "NX"?

### ISSUE THREAD

**Issue by mikedep333** [ArcticaProject/nx-libs#5]
If you look at debian/control, you will see package descriptions like:

> Package: libxcompshad3
> ...
> Description: nx-X11 shadowing library
> NX is a software suite which implements very efficient
> compression of the X11 protocol. This increases performance when
> using X applications over a network, especially a slow one.
> .
> This package provides a library for shadow session support.

Should we continue using the term "NX"? Or should we switch all references to "nx-libs"? Or perhaps replace both NX and nx-libs entirely with some other name like ["LBX2"](http://keithp.com/~keithp/talks/lbxpost/)?

**Comment by mikedep333** [comment]
@gznget @sunweaver @nitomartinez @Ionic 

I propose `Xccc`:

X11 Cached, Compressed and short-Circuited

Here is a proposed site design:
http://mikedep333.github.io/Xccc/
And the code for it:
https://github.com/mikedep333/Xccc

I believe that we need to rename nx-libs quickly (but not necessarily on the 3.5.0.x branch.)
Further reasoning is in the proposed site design.

**Comment by sunweaver** [comment]
How about Xc3? (Not that CCC is Chaos Computer Club in Germany/Europe, so there is a slight name clashing...).

**Comment by mikedep333** [comment]
I was wondering that myself. I thought that "CCC" sounded familiar.

Some other piece of software is already called ["XCC".](https://developer.marklogic.com/products/xcc) But by using "Xc3", we can put the project name in the binary names (Xc3proxy and Xc3agent).

**Comment by Ionic** [comment]
I don't think a digit in the name is a great idea. Can easily be confused with a version number or other stuff.

I'd rather look for something like "NAX" (Network Accelerated X) or the like.

**Comment by mikedep333** [comment]
I tend to agree with @Ionic . It can also create confusion with "X11" or "X2Go", but nx-libs is not specific to X2Go.

I still think Xccc is best because it highlights the 3 things that nx-libs does, but VNC only does 1 of (compression.) But my 2nd favorite so far is NAX.

If only the acronym "WAX" (WAN Accelerated X) didn't have a bad connotation ([Wax](%28https://en.wikipedia.org/wiki/Wax%29) is sticky, which means it is not fast moving. Also, "ear wax.").

**Comment by Ionic** [comment]
X11 actually is called X11 because it's the 11th protocol version. :)
(Although admittedly nowadays it's its own trademark.)

X2Go is "kind of okay" due to the the digit being surrounded by other letters.

WAX... there was also VAX, so...

I'm trying to come up with something involving "X", "accelerated" or "enhanced" and "network" (or similar), but haven't had a brilliant idea yet. Network enhanced X Technology? Better not. ;)

**Comment by mikedep333** [comment]
@Ionic, yeah.

I am going to look through dictionary files for either a name or inspiration. Not that I have given up on Xccc, but there may be a better name.

Also, do not forget this lesson on naming software (or games):
[The word Narbacular, which does not exist in any dictionary, was chosen primarily to aid in internet search engine results.](https://en.wikipedia.org/wiki/Narbacular_Drop)

**Comment by sunweaver** [comment]
@mikedep333: Let's make sure that the new name is easily pronouncable and gives a fine melody when saying it. Xccc is a little tongue twister. Do we necessarily need to stay below 8 chars for the executables?

How about these, may: Xccc (nxagent), x3cproxy?

**Comment by sunweaver** [comment]
actually, xc3proxy...

**Comment by mikedep333** [comment]
Hi @sunweaver,

It is true that Xccc is a bit long to say. However, I liked Xccc because it highlights the 3 things we do (Caching, Compression and short-Circuiting.) And "short-Circuting" (a term I **made up** for what we currently do to eliminate round-trips) is also a fine excuse for the executable having "Xcc" in the name.

Like I said, I intend to look through the dictionary files for more inspiration. I am open to other suggestions, including other names, and including adopting 1 name but having different names for the executables (like you suggested.) The terms "agent" and "proxy" are themselves confusing.

Even if we do adopt a different name, it may be a good idea to use the slogan "X11 Cached, Compressed and short-Circuited".

**Comment by nitomartinez** [comment]
My two cents:

NXC -> New X11 Channel Protocol

Pros:
- It maintains the current NX name showing some history behind it
- It uses the C which you seem to like.

**Comment by sunweaver** [comment]
I would like to bring up this discussion once more, so we maybe can quickly close this issue report.

As we are currently moving forward towards inclusion of nx-libs in all the major distros, I vote against a name change. At least not at this point or any near future point.

The term is coined and is associated with low bandwidth X11 connections. People have noticed, that we have been doing a shitload of work to clean up the code.

Enforcing a name change now would mean a considerable bunch of work. We should have done that 2 years back.

Anyone objecting to me closing this issue report? (If not, I will close it in two days from now).

**Comment by sunweaver** [comment]
@Ionic: @uli42: @mikedep33: @stefanbaur: @narenas: please take notice of https://github.com/ArcticaProject/nx-libs/issues/5#issuecomment-368794327

And comment, if needed... Thanks!

**Comment by sunweaver** [comment]
@mikedep333:  please take notice of #5 (comment)

And comment, if needed... Thanks!

**Comment by stefanbaur** [comment]
cf. https://en.wikipedia.org/wiki/Bicycle_shed_effect

TL;DR: Keep the name and close this report; or at least postpone the decision until the next major release.

**Comment by Ionic** [comment]
Rebranding would take a lot of additional (packaging) work, so... probably not wise.

**Comment by Ionic** [comment]
> cf. https://en.wikipedia.org/wiki/Bicycle_shed_effect

Do we have another entry? BSE?
